utils/editSubscriber.tsx

```diff
@@ -74,40 +63,34 @@ const updateSubscriber = async (subscriberData: any, token: string) => {
     existingSubscriberData["AuthenticationSubscription"]["permanentKey"]["permanentKeyValue"] = subscriberData.key;
     existingSubscriberData["AuthenticationSubscription"]["sequenceNumber"] = subscriberData.sequenceNumber;
 
-    const updateSubscriberResponse = await apiPostSubscriber(subscriberData.UeId, subscriberData, token);
-    if (!updateSubscriberResponse.ok) {
-      throw new Error(
-        `Error editing subscriber. Error code: ${updateSubscriberResponse.status}`,
-      );
-    }
+    await apiPostSubscriber(subscriberData.UeId, subscriberData, token);
+
   } catch (error) {
-    console.error(error);
+    console.error(`Failed to update subscriber ${subscriberData.UeId} : ${error}`);
+    throw error;
   }
 }
 
-const getDeviceGroupData = async (deviceGroupName: string, token: string) => {
-  try {
-    const existingDeviceGroupResponse = await apiGetDeviceGroup(deviceGroupName, token);
-    var existingDeviceGroupData = await existingDeviceGroupResponse.json();
-
-    if (existingDeviceGroupData && !existingDeviceGroupData["imsis"]) {
-      existingDeviceGroupData["imsis"] = [];
+async function updateOldDeviceGroup(imsi: string, oldDeviceGroupName: string, token: string) : Promise<void>{
+  try{
+    if (!oldDeviceGroupName){
+      return
+    }
+    var oldDeviceGroupData = await getDeviceGroup(oldDeviceGroupName, token);
+    if (oldDeviceGroupData){
+      const index = oldDeviceGroupData["imsis"].indexOf(imsi);
+      if (index !== -1) {
+        oldDeviceGroupData["imsis"].splice(index, 1);
```

Comment:
```suggestion
        oldDeviceGroupData.imsis.splice(index, 1);
```
We can simlify it as `imsis` should always exists because of the `getDeviceGroup`definition.